### PR TITLE
Deletes entries in CHANGES directory after creatinga new Y branch

### PR DESCRIPTION
**Pull request by dkliban**
Also stops opening changelog PRs after a release.

fixes: #754

### CODE REVIEW

File: templates/github/.github/workflows/create-branch.yml.j2 [hunk 1]
```diff
@@ -52,7 +52,11 @@ jobs:
           git checkout {{ plugin_default_branch }}
           bump2version --no-commit minor
 
-      - name: Make a PR with version bump
+      - name: Remove entries from CHANGES directory
+        run: |
+          git rm CHANGES/*
```
Comment (gerrod3) [inline templates/github/.github/workflows/create-branch.yml.j2:57]: I think this also deletes the dot files and the CHANGES folder itself.

Comment (gerrod3) [reply templates/github/.github/workflows/create-branch.yml.j2:57]: Have you tested the command on your machine? On mine in an 'empty' plugin CHANGES directory it deleted both dot files and the directory itself. Maybe this isn't a fair comparison if we expect there to always be changelog entries present.

```
[gerrod@localhost pulp_python]$ git rm CHANGES/*
rm 'CHANGES/.TEMPLATE.rst'
rm 'CHANGES/.gitignore'
[gerrod@localhost pulp_python]$ ls -la CHANGES/
ls: cannot access 'CHANGES/': No such file or directory
```

Comment (dkliban) [reply templates/github/.github/workflows/create-branch.yml.j2:57]: Your shell has globstar enabled. I'll improve the patch to be more specific and only remove entries.

File: templates/github/.github/workflows/create-branch.yml.j2 [hunk 2]
```diff
@@ -52,7 +52,11 @@ jobs:
           git checkout {{ plugin_default_branch }}
           bump2version --no-commit minor
 
-      - name: Make a PR with version bump
+      - name: Remove entries from CHANGES directory
+        run: |
+          find CHANGES -type f -regex ".*\.\(bugfix\|doc\|feature\|misc\)" -exec git rm {} +
```
Comment (gerrod3) [inline templates/github/.github/workflows/create-branch.yml.j2:57]: There's two more: removal & deprecation. https://github.com/pulp/pulpcore/blob/main/pyproject.toml#L20-L50

**Review by gerrod3 — CHANGES_REQUESTED** [review]

File: templates/github/.github/workflows/create-branch.yml.j2 (hunk 2 above)
Comment (dkliban) [reply templates/github/.github/workflows/create-branch.yml.j2]: Great eye! I added both!

File: templates/github/.github/workflows/release.yml.j2 [hunk 3]
```diff
@@ -285,19 +285,3 @@ jobs:
 
       - name: Stage changelog for {{ plugin_default_branch }} branch
         run: python .github/workflows/scripts/stage-changelog-for-default-branch.py {{ "${{ github.event.inputs.release }}" }}
```
Comment (gerrod3) [inline templates/github/.github/workflows/release.yml.j2:287]: I think this step, the one above it (Cleanup repository before making changelog PR) and the workflow script (stage-changelog-for-default-branch.py) can all be removed as well.

**Review by gerrod3 — APPROVED** [review]
@dkliban Nice! One less PR to manually review.

File: templates/github/.github/workflows/create-branch.yml.j2 (hunk 2 above)
Comment (mdellweg) [reply templates/github/.github/workflows/create-branch.yml.j2]: Actually, there is a default set or it is configured in pyproject.yaml. How much dynamic do we want to make it?

**Review by mdellweg — APPROVED** [review]

**Pull request merged**